kubelet/cm/cpumanager: Improving test coverage

tmilox opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

/kind cleanup


#### What this PR does / why we need it:

If applied this commit will increase the unit tests code coverage of `pkg/kubelet/cm/cpumanager`.

- Initial unit tests code coverage

<img width="733" alt="Screenshot 2023-01-02 at 20 34 22" src="https://user-images.githubusercontent.com/9576370/210363360-b1588b26-b89b-4dfa-a388-b515cb9fb935.png">

- After Changes

<img width="732" alt="Screenshot 2023-01-03 at 14 07 09" src="https://user-images.githubusercontent.com/9576370/210363396-5229a00d-8788-4a87-9359-8ac241de05cc.png">


#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

TommyStarK commented:
/kind cleanup
/area kubelet
/sig node

TommyStarK commented:
Hello @bart0sh 

I opened this one but the robot failed to label it properly. Had to do it manually. Not sure if I should add the size myself for example or if I can trigger the label step again. 

Happy new year, best wishes. 
Cheers,

bart0sh commented on pkg/kubelet/cm/cpumanager/policy_options_test.go:
```diff
@@ -109,53 +109,75 @@ func TestPolicyOptionsAvailable(t *testing.T) {
 			topology:      topoDualSocketMultiNumaPerSocketHT,
-			topoMgrPolicy: topologymanager.PolicySingleNumaNode,
+			topoMgrPolicy: topologymanager.PreferClosestNUMANodes,
 			expectedErr:   true,
```
`expectedValidateErr` is missing here. You should either put it everywhere or only use it when it's value is `true`.

TommyStarK replied:
My bad, sorry. Resolved in 832cb5ad9e3

bart0sh commented:
/ok-to-test
/priority important-longterm
/triage accepted

bart0sh commented:
/check-required-labels

bart0sh commented:
@TommyStarK 
> Not sure if I should add the size myself for example or if I can trigger the label step again.

I hope the robot will relabel this PR when it's rebased.

bart0sh commented:
/assign

bart0sh commented:
/release-note-none

TommyStarK commented:
/retest-required

bart0sh commented on pkg/kubelet/cm/cpumanager/policy_options_test.go:
```diff
@@ -168,10 +185,13 @@ func TestValidateStaticPolicyOptions(t *testing.T) {
-			policyOpt, _ := NewStaticPolicyOptions(testCase.policyOption)
-			err := ValidateStaticPolicyOptions(policyOpt, testCase.topology, topoMgrStore)
+			policyOpt, err := NewStaticPolicyOptions(testCase.policyOption)
+			if err == nil && testCase.expectedErr {
```
This check misses the case with ```expectedErr = false``` and ```err != nil```. Can you use the same approach as with ```expectedValidateErr``` below?

TommyStarK replied:
Sure thing, done in 1578af56b87

TommyStarK commented:
@bart0sh Do you want me to squash my commits into a single one ?

bart0sh commented:
@TommyStarK it would be nice, but you don't have to. `tide/merge-method-squash` should do it.
Please, read [this guide](https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#squashing) for more details.

bart0sh commented:
/lgtm
/assign @derekwaynecarr @dchen1107 @mrunalp

bart0sh commented:
/assign @klueska

TommyStarK commented:
@bart0sh It's done. Thank you as usual for taking the time and sharing this guide I obviously missed

bart0sh commented:
/lgtm

TommyStarK commented:
/assign @derekwaynecarr

tmilox committed:
kubelet/cm/cpumanager: Improving test coverage

topology: add unit tests for CPUsPerCore and CPUsPerSocket
topology: fix variable declaration with assignment on next line (S1021)go-staticcheck
state: cover GetCPUSetOrDefault in unit tests
policy: add unit tests for static policy and policy options

Signed-off-by: TommyStarK <thomasmilox@gmail.com>

TommyStarK commented:
@bart0sh PR has been rebased

bart0sh commented:
/lgtm
@klueska can you have a look?

TommyStarK commented:
@derekwaynecarr I know you are super busy, another small coverage improvement

TommyStarK commented:
/remove-lifecycle stale

TommyStarK commented:
@klueska when you have time I'd like to complete this PR and merge it
